src/libAtomVM/externalterm.c

```diff
@@ -355,6 +402,22 @@ static term parse_external_terms(const uint8_t *external_term_buf, int *eterm_si
             return term_make_maybe_boxed_int64(ctx, value);
         }
 
+        case SMALL_BIG_EXT: {
+            uint8_t num_bytes = external_term_buf[1];
+            uint8_t sign = external_term_buf[2];
+            avm_uint64_t unsigned_value = read_bytes(external_term_buf + 3, num_bytes);
+            // NB due to call to calculate_heap_usage:
+            // 1. 0 <= unsigned_value <= INT64_MAX if sign is 0
+            // 2. 0 <= unsigned_value <= INT64_MAX + 1 if sign is not 0
+            avm_int64_t value = 0;
+            if (sign != 0x00) {
+                value = -((avm_int64_t) unsigned_value);
+            } else {
+                value = (avm_int64_t) unsigned_value;
+            }
+            return term_make_maybe_boxed_int64(ctx, value);
```

Comment:
I am confused why we are not updating `eterm_size` here.

Reply:
Good catch.